lms/djangoapps/courseware/tests/test_video_mongo.py

```diff
@@ -814,52 +814,71 @@ class TestVideoCDNRewriting(BaseTestXmodule):
         """
         Test successful CDN request.
         """
-        cdn_response_video_url = settings.CDN_VIDEO_URLS["CN"] + self.original_video_file
+        cdn_response_video_url = settings.VIDEO_CDN_URL["CN"] + self.original_video_file
 
         self.assertEqual(
-            rewrite_video_url(settings.CDN_VIDEO_URLS["CN"], self.original_video_url),
+            rewrite_video_url(settings.VIDEO_CDN_URL["CN"],
+                              self.original_video_url,
+                              settings.VIDEO_CDN_REWRITABLE_SOURCE_DOMAINS),
             cdn_response_video_url
         )
 
-    @patch.dict("django.conf.settings.CDN_VIDEO_URLS",
-                {"CN": "https://chinacdn.cn/"})
+    @override_settings(VIDEO_CDN_REWRITABLE_SOURCE_DOMAINS=["www.originalvideo.com"])
+    @override_settings(VIDEO_CDN_URL={"CN": "https://chinacdn.cn/"})
     def test_rewrite_url_concat(self):
         """
         Test that written URLs are returned clean despite input
         """
-        cdn_response_video_url = settings.CDN_VIDEO_URLS["CN"] + "original_video.mp4"
+        cdn_response_video_url = settings.VIDEO_CDN_URL["CN"] + "original_video.mp4"
 
         self.assertEqual(
-            rewrite_video_url(settings.CDN_VIDEO_URLS["CN"] + "///", self.original_video_url),
+            rewrite_video_url(settings.VIDEO_CDN_URL["CN"] + "///",
+                              self.original_video_url,
+                              settings.VIDEO_CDN_REWRITABLE_SOURCE_DOMAINS),
             cdn_response_video_url
         )
 
+    @override_settings(VIDEO_CDN_REWRITABLE_SOURCE_DOMAINS=["www.originalvideo.com"])
+    @override_settings(VIDEO_CDN_URL={"CN": "https://chinacdn.cn/"})
     def test_rewrite_video_url_invalid_url(self):
         """
         Test if no alternative video in CDN exists.
         """
         invalid_cdn_url = 'http://http://fakecdn.com/'
-        self.assertIsNone(rewrite_video_url(invalid_cdn_url, self.original_video_url))
+        self.assertIsNone(rewrite_video_url(invalid_cdn_url,
+                                            self.original_video_url,
+                                            settings.VIDEO_CDN_REWRITABLE_SOURCE_DOMAINS))
+
+    @override_settings(VIDEO_CDN_REWRITABLE_SOURCE_DOMAINS=["oktorewrite.com"])
+    @override_settings(VIDEO_CDN_URL={"CN": "https://chinacdn.cn/"})
+    def test_rewrite_video_url_whitelist(self):
+        """
+        Test if the original URL is not on the rewrite whitelist
+        """
+        self.assertIsNone(rewrite_video_url(settings.VIDEO_CDN_URL["CN"],
+                                            self.original_video_url,
+                                            settings.VIDEO_CDN_REWRITABLE_SOURCE_DOMAINS))
 
     def test_none_args(self):
         """
         Ensure None args return None
         """
-        self.assertIsNone(rewrite_video_url(None, None))
+        self.assertIsNone(rewrite_video_url(None, None, None))
 
     def test_emptystring_args(self):
```

Comment:
Could: I think you should add a test for a list with multiple entries in the rewritable domains.